SConstruct

```diff
@@ -292,6 +288,25 @@ opts.Add("linkflags", "Custom flags for the linker")
 # in following code (especially platform and custom_modules).
 opts.Update(env)
 
+# Process user profiles with custom arguments.
+custom_profiles = env["profile"].split(",")
+profiles_to_load = []
+custom_args = {}
+for profile in custom_profiles:
+    if os.path.isfile(profile):
+        profiles_to_load.append(profile)
+    elif os.path.isfile(profile + ".py"):
+        profiles_to_load.append(profile + ".py")
+if profiles_to_load:
+    for profile in profiles_to_load:
+        profile_args = {}
+        with open(profile, "r", encoding="utf-8") as f:
+            code = f.read()
+            exec(code, {}, profile_args)
+        custom_args = {**custom_args, **profile_args}
+    # Update the variables based on the profiles.
+    opts.Update(env, {**ARGUMENTS, **custom_args, **env})
```

Comment:
Might need to reason a bit on the order of merging overlapping arguments from command line, `custom.py`, and then potential further additions to the env. Will need to do more testing on this PR.

Reply:
I think that cli args should override profile args. Cli args + profile args = user args. User args should override env... Your current code does exactly opposite (prefers env over profile args over command line args)

And I would suggest to have separate variable for storing both cli and profile args (`user_args` or something like it). It will be handy in several places. And IIUC whole `if (key not in ARGUMENTS or ARGUMENTS[key] == "auto") and key not in custom_args:` could be then written shorter with `user_args`